clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/ActiveActiveStoreIngestionTask.java

```diff
@@ -234,28 +234,29 @@ private byte[] prependReplicationMetadataBytesWithValueSchemaId(ByteBuffer metad
    * @param subPartition The partition to fetch the replication metadata from storage engine
    * @return The object containing RMD and value schema id. If nothing is found, return {@code Optional.empty()}
    */
-  private Optional<RmdWithValueSchemaId> getReplicationMetadataAndSchemaId(
+  protected Optional<RmdWithValueSchemaId> getReplicationMetadataAndSchemaId(
       PartitionConsumptionState partitionConsumptionState,
       byte[] key,
       int subPartition) {
+    double replicationMetadataLatency = 0; // Initialize default latency to 0 so that we always generate a metric
+    RmdWithValueSchemaId rmd = null;
     PartitionConsumptionState.TransientRecord cachedRecord = partitionConsumptionState.getTransientRecord(key);
     if (cachedRecord != null) {
       hostLevelIngestionStats.recordIngestionReplicationMetadataCacheHitCount();
-      return Optional.of(
-          new RmdWithValueSchemaId(
-              cachedRecord.getValueSchemaId(),
-              rmdProtocolVersionID,
-              cachedRecord.getReplicationMetadataRecord()));
-    }
-
-    final long lookupStartTimeInNS = System.nanoTime();
-    byte[] replicationMetadataWithValueSchemaBytes = storageEngine.getReplicationMetadata(subPartition, key);
-    hostLevelIngestionStats
-        .recordIngestionReplicationMetadataLookUpLatency(LatencyUtils.getLatencyInMS(lookupStartTimeInNS));
-    if (replicationMetadataWithValueSchemaBytes == null) {
-      return Optional.empty(); // No RMD for this key
```

Comment:
This tweak couldn't resolve the problem. This function 'getReplicationMetadataAndSchemaId' is only called when processing incoming writes; if there is no incoming writes, there will still be gaps in the metric; by gaps, I mean no metric value at all, not 0 (the current problem)

Reply:
Hmmm, in the existing code path , a lot of the lookups will go through cache and no stats will be published. This will definitely fix that. If we did not have any traffic , then the latency of lookup doesn't really have meaning. It will be hacky to publish dummy metrics which don't have any meaning. What will be the hook to publish ? How many such metrics need to be published?

Reply:
Can you remind me why we must have default latency metrics as 0? Given not all of code paths are examined, it's probably fine to have no emission for some metrics.

Besides, that's `hostLevelIngestionStats` so unless there's no AA task that can ingest data on this host, which sounds very unlikely to me, this metric must have some data points, correct?